streams/test-utils/src/main/java/org/apache/kafka/streams/TopologyTestDriver.java

```diff
@@ -439,6 +441,10 @@ private void captureOutputRecords() {
         // Capture all the records sent to the producer ...
         final List<ProducerRecord<byte[], byte[]>> output = producer.history();
         producer.clear();
+        if (eosEnabled) {
```

Comment:
We also need to check if the producer is already closed here.
```
if (eosEnabled && !producer.closed()) {
```

Reply:
Fixed